index.js

```diff
@@ -977,23 +1001,45 @@ module.exports = {
     };
 
     // Invoke a remote API. A simple wrapper around request-promise
-    // build the correct URL. `options` is the usual `request` options object.
+    // build the correct URL. `requestOptions` is the usual `request` options object.
+    // `options` may contain `deployTo`; when the `deployTo` option for the
+    // module is an array, `options.deployTo` must match the `name` property of one
+    // server in that array.
+    //
     // Returns a promise.
-    self.remoteApi = function(verb, options) {
-      var deployTo = self.options.deployTo;
-      if (!deployTo) {
-        return Promise.reject(new Error('deployTo option must be configured'));
-      }
+
+    self.remoteApi = function(verb, requestOptions, options) {
+      var deployTo = self.resolveDeployTo(options);
       if (!deployTo.apikey) {
         return Promise.reject(new Error('deployTo.apikey option must be configured'));
       }
-      var options = _.merge({
+      requestOptions = _.merge({
         headers: {
           'Authorization': 'Bearer ' + deployTo.apikey
         }
-      }, options);
+      }, requestOptions);
       var url = deployTo.baseUrl + deployTo.prefix + '/modules/apostrophe-review-and-deploy/' + verb;
-      return request(deployTo.baseUrl + deployTo.prefix + '/modules/apostrophe-review-and-deploy/' + verb, options || {});
+      return request(deployTo.baseUrl + deployTo.prefix + '/modules/apostrophe-review-and-deploy/' + verb, requestOptions);
+    };
+
+    // If the module's `deployTo` option is an object with properties such
+    // as apikey, etc., returns that object directly. If the module's
+    // `deployTo` option is an array, returns the entry whose `name`
+    // property matches the `deployTo` property of the `options` object
+    // passed to this method.
+    
+    self.resolveDeployTo = function(options) {
+      var deployTo = self.options.deployTo;
+      if (!deployTo) {
```

Comment:
Hey doofus, you can't reject a promise from this synchronous function. Fix that.